dts/msm8939-qrd.dts

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+/include/ "msm8916.dtsi"
+
+/ {
+	qcom,msm-id = <239 0>;
+	qcom,board-id = <0x1000b 5>;
+
+	lenovo-pb1770m {
+		model = "Lenovo PHAB Plus";
+		compatible = "lenovo-pb1770m", "lenovo-pb1770n", "qcom,msm8939-qrd-skuk", "qcom,msm8939-qrd", "qcom,msm8939", "qcom,qrd", "lk2nd,device";
```

Comment:
You don't need all of them. You need to pick one for your device, e.g. `lenovo,pb1770m` or maybe `lenovo-pb1770` if the two are similar enough. Then you should add `qcom,msm8939` and `lk2nd,device`, e.g.

```suggestion
		compatible = "lenovo,pb1770m", "qcom,msm8939", "qcom,qrd", "lk2nd,device";
```

Also, it seems like this is a shared DTS (in other words, there could be other devices with the same `qcom,board-id`.

This means that you need to add something like `lk2nd,match-cmdline` to make sure only your device is matched. Like here:

https://github.com/msm8916-mainline/lk2nd/blob/799945cb0d2871cf404b4a137637f6ab9d052f49/dts/msm8939-mtp.dts#L12-L16

Reply:
There's not a "PB1770", but PB1-770M is just the global variant while PB1-770N is the china only one, and only difference afaik is firmware.